moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1960,7 +1927,7 @@ double RobotState::computeCartesianPath(const JointModelGroup* group, std::vecto
 
     // Explicitly use a single IK attempt only: We want a smooth trajectory.
     // Random seeding (of additional attempts) would probably create IK jumps.
-    if (setFromIK(group, pose, link->getName(), 1, 0.0, validCallback, options))
+    if (setFromIK(group, pose, link->getName(), 0.0, validCallback, options))
```

Comment:
I don't like this at all. This will have non-trivial consequences for users of the computeCartesianPath method. The single attempt, (as you pointed out to me) tries to keep the IK solver from choosing random seeds which would otherwise create unsafe paths

Reply:
... which I thought now is ensured by a zero timeout. But you are right, it isn't. The zero timeout is replaced by the default timeout later on...
What we actually need to use here is another call to the IK solver. Please see #1285 for a discussion of API ambiguities.

Reply:
I resolved this in a two-fold fashion: 
* Passing an epsilon timeout (instead of 0.0), should ensure that only a single attempt can be processed as originally intended. However, this is somehow dirty. It would be better to clarify the semantics of IK calls as discussed in #1285 and use an appropriate function that explicitly searches for the closest solution to the seed.
* Passing in consistency_limits if corresponding joint-space jump limits are defined (which is not generally the case as these parameters default to zero).

Reply:
Please disregard the previous comments. Everything was just fine:
The `attempt=1` parameter, originally passed to `setFromIK` was only handled in `setFromIK` and never passed onto the IK solver. IK solvers didn't and still don't have a notion of max attempts.
Removing the notion of max attempts from `setFromIK` with this PR, there is no need to pass a 1 here as well, as only a single attempt is made anyway.
It remains the fact that the IK solver still can try several attempts (with random seeds) - as before.
The only available API to limit joint-space deviations from the seed is to pass consistency limits.
However, as this is not related to this PR, I filed this patch as a separate PR: #1293.